EKF/ekf_helper.cpp

```diff
@@ -852,72 +861,172 @@ void Ekf::calcEarthRateNED(Vector3f &omega, float lat_rad) const
 	omega(2) = -CONSTANTS_EARTH_SPIN_RATE * sinf(lat_rad);
 }
 
-// gets the innovations of velocity and position measurements
-// 0-2 vel, 3-5 pos
-void Ekf::get_vel_pos_innov(float vel_pos_innov[6])
+void Ekf::getGpsVelPosInnov(float hvel[2], float &vvel, float hpos[2],  float &vpos)
+{
+	memcpy(hvel, _gps_vel_pos_innov+0, sizeof(float) * 2);
```

Comment:
Returning matrix::Vectors is also fine.

Reply:
can you give an example how you mean that?

Reply:
Actually this one isn't a great example, but in other cases we could easily/safely return a Vector by value rather than passing in an array or pointer to fill.

Examples

``` C++
    // Current PX4/ecl:
    // we often pass in a pointer to fill with 1, 2, or 3 floats
    void estimator_interface::get_velocity(float *vel);
    // wind velocity pass in vector to fill with 2 floats
    void Ekf::get_wind_velocity(float *wind);


    // SUGGESTION:
    // whenever possible we should prefer simple and safe getters that return by value
    matrix::Vector3f estimator_interface::get_velocity() const;
    matrix::Vector2f Ekf::get_wind_velocity() const;

```

I'm mainly trying to say don't be afraid to return these thing as values. It's easier to use, harder to misuse, and not going to be any slower.